### ISSUE TITLE
[BUG]: OOM--Training reward model

### ISSUE THREAD

**Issue by shuibuxing0023** [hpcaitech/ColossalAI#4360]
### 🐛 Describe the bug

![image](https://github.com/hpcaitech/ColossalAI/assets/107054070/a8f85a94-8c95-4aad-b5df-74109c62db72)
Stage1 can be successfully performed in this way, but if I open the grad_checkpoint option, there will still be OOM problems;
Stage2 also had OOM problems, the following are the commands I use

`torchrun --standalone --nproc_per_node=1 train_reward_model.py  --pretrain "/root/autodl-tmp/output/coati-7B"  --model 'llama'  --strategy ddp  --loss_fn 'log_exp' --save_path "/root/autodl-tmp/output/coati-7B-pt/rmstatic.pt" --batch_size 1   --max_epochs 1  --lora_rank 1`

### Environment

CUDA：1*3090

**Comment by CWHer** [comment]
FYI, I run this command on A100, and `torch.cuda.max_memory_allocated()` yields 
```
[INFO]: Peak memory usage: 23653.14 MiB
```
